### PR TITLE
As a Student, I can edit/view information in my profile

### PR DESCRIPTION
**Pull request by abdimohamud**
StudentDemo
- Implemented Student view of profile and feature to edit it, when logging in with "mentee" role.
- copied layout and view of headmaster, but restricted to just being able to navigate to profile, but landing and other features or pages will be added in via src/component/Student folder 
- state is being handled in the same flow as teacher with separate reducers and actions 
- added basic test for rendering dashboard and checking to see if the welcome message is correct for the user role.

Viewable values from a student:
       - first_name
       - last_name
       - email
       - primary_language
       - dob
       - mentee_picture
       - english_lvl
       - math_lvl
       - reading_lvl
       - school_lvl
       - academic_description
       - support_needed

# Reviewing a PR
- [x] Leave comments and suggestions!
- [x] Is code formatted properly?
- [x] Is commented out code deleted?
- [x] Are functions relatively small?
- [x] Are console logs removed?
- [x] Is there duplicate code?
- [x] Does code have TODO's commited to main? (please don't)

## Type of Change
- [x] New feature
- [ ] Refactoring / Bug fix
- [ ] Documentation update 
​

## Requester Questions

```Add any questions, comments, or concerns you have as the person initiating the pull request for your reviewers```
- used mocked data for view and edit, will be taking a look at back-end later today to find how to implement a put request based on the id, role, and user.
- switched to a new branch for naming conventions and was confused initially on which form I was editing so will scrape or use functionality in other task (edit/view-student-profiles)

### CODE REVIEW

File: src/__tests__/StudentDashboard.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import reducer from '../state/reducers';
+import StudentDashboard from '../components/pages/Student/StudentDashboard';
+
+describe('Render Student Dashboard', () => {
+  test('it should mount a div based on props', () => {
+    const store = createStore(reducer, applyMiddleware(thunk));
+    const { container, getByText } = render(
+      <Provider store={store}>
```
Comment (stackpearson) [inline src/__tests__/StudentDashboard.test.js:14]: Implementing testing, nice! I'll need to make a few tests for the mentee page I forgot about that.

File: src/components/pages/Student/StudentProfile/ProfileForm.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+  availability: {
+    time_zone: '',
+    as_early_as: '',
+    as_late_as: '',
+    methods: [],
+  },
+  dynamic_questions: [],
+};
+
+const dateFormat = 'MM/DD/YYYY';
```
Comment (stackpearson) [inline src/components/pages/Student/StudentProfile/ProfileForm.js:45]: Same date format I've seen on other pages, thanks for keeping it consistent.

File: src/components/pages/Student/StudentProfile/ProfileForm.js [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    setValue(e.target.value);
+  };
+
+  const handleSubmit = async () => {
+    console.log(formData);
+    props.editStudentProfile(params, formData);
+  };
+
+  const handleChange = e => {
+    debugLog(formData);
```
Comment (stackpearson) [inline src/components/pages/Student/StudentProfile/ProfileForm.js:76]: Any idea how this debugLog works? I've seen it in a few pages but haven't pieced it together yet.

**Review by stackpearson — APPROVED** [review]
Solid, looks like this meets all the needs of the user story!

File: src/components/pages/Student/StudentProfile/ProfileForm.js (hunk 3 above)
Comment (abdimohamud) [reply src/components/pages/Student/StudentProfile/ProfileForm.js:76]: I was confused on why that was there as well, but I think its a placeholder for redux logger, so it shows the actions be used and state changing. Will look into it more and see why we arent using redux logger and let you know 
image

Comment (stackpearson) [reply src/components/pages/Student/StudentProfile/ProfileForm.js:76]: Ahh yeah, that has to be it. I'll dig into it a bit more too and share if I find anything.

File: src/components/pages/Student/StudentDashboard.js [hunk 4]
```diff
@@ -93,4 +93,4 @@ function StudentDashboard() {
 
 export default StudentDashboard;
 
-// cloned same format as headmaster dashboard with only visible profile view and ability to edit
+// cloned same format as headmaster dashboard with only visible profile view and ability to edit profile
```
Comment (tompsherman) [inline src/components/pages/Student/StudentDashboard.js:96]: it looks like theres multiple types of comment styles used throughout the files, we should maybe standardize whether we use {* *} or //

**Review by tompsherman — APPROVED** [review]

**Pull request merged**